### PR TITLE
New tool 2dCompImage

### PR DESCRIPTION
**Pull request by kerautret**
# PR Description
New tools to compare 2D images (and visualize errors).

# Checklist

- [x] Doxygen documentation of the code completed (classes, methods, types, members...).
- [x] Main tool doxygen documentation (following existing documentation of [DGtalTools documentation](http://dgtal.org/doc/tools/nightly/).
- [x] Check if it follows the tools structure described in [CONTRIBUTING.md](https://github.com/DGtal-team/DGtalTools/blob/master/CONTRIBUTING.md)
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtalTools/blob/master/ChangeLog.md) added.
- [x] Update the readme with potentially a screenshot of the tools if it applies. 
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).

### CODE REVIEW

File: doc/visualization.dox [hunk 1]
```diff
@@ -19,7 +19,7 @@
 @section visualization_Doc Visualization
 
 
-
+  - @ref Doc2dCompImage : Computes and displays image comparisons (squared and absolute differences).
```
Comment (dcoeurjo) [inline doc/visualization.dox]: Compare images and display differences (squared and absolute differences).

?

File: visualisation/2dCompImage.cpp [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+ @b Allowed @b options @b are :
+ 
+ @code
+  -h [ --help ]                    display this message
+  -a [ --imageA ] arg              Input filename of image A.
+  -b [ --imageB ] arg              Input filename of image B.
+  -e [ --imageError ] arg          Output error image basename (will generate 
+                                   two images <basename>MSE.ppm and 
+                                   <basename>MAE.ppm).
+  -S [ --fixMaxColorValueMSE ] arg Fix the maximal color value for the scale
```
Comment (dcoeurjo) [inline visualisation/2dCompImage.cpp]: Fix -> set?

File: visualisation/2dCompImage.cpp [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+
+/**
+ * Compute statistics on the mean absolute difference between pixel values of image A and B. 
+ **/
+Statistic< int>
+getMAEstats(const Image2D & imageA, const Image2D &imageB, Image2DErr &imageMAE)
+{
+  Statistic< int> stat(false);
+  for(Image2D::Domain::ConstIterator it = imageA.domain().begin();
```
Comment (dcoeurjo) [inline visualisation/2dCompImage.cpp]: or more simply: 

`for(auto point: imageA.domain())`

Comment (dcoeurjo) [reply visualisation/2dCompImage.cpp]: ;)

File: visualisation/2dCompImage.cpp [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+int main( int argc, char** argv )
+{
+  // parse command line -------------------------------------------------------
+  po::options_description general_opt("Allowed options are");
+  general_opt.add_options()
+    ("help,h", "display this message")
+    ("imageA,a", po::value<std::string >(), "Input filename of image A." )
+    ("imageB,b", po::value<std::string >(), "Input filename of image B." )
+    ("imageError,e", po::value<std::string >(), "Output error image basename (will generate two images <basename>MSE.ppm and <basename>MAE.ppm)." )
+    ("fixMaxColorValueMSE,S", po::value<int>(), "Fix the maximal color value for the scale display of MSE (else the scale is set the maximal MSE value)." )
```
Comment (dcoeurjo) [inline visualisation/2dCompImage.cpp]: Fix -> set ?

File: doc/visualization.dox (hunk 1 above)
Comment (kerautret) [reply doc/visualization.dox]: yes better thanks ;)

File: visualisation/2dCompImage.cpp (hunk 2 above)
Comment (kerautret) [reply visualisation/2dCompImage.cpp]: yes

File: visualisation/2dCompImage.cpp (hunk 3 above)
Comment (kerautret) [reply visualisation/2dCompImage.cpp]: yes thanks C++11 ;)

**Comment by kerautret** [comment]
all done @dcoeurjo thanks for the review

**Comment by dcoeurjo** [comment]
perfect. thanks.

**Pull request merged**